src/tools/miri/tests/fail/validity/dyn-trait-leak-check.rs

```diff
@@ -0,0 +1,41 @@
+// Test that transmuting from `&dyn Dyn<fn(&'static ())>` to `&dyn Dyn<for<'a> fn(&'a ())>` is UB.
+//
+// The vtable of `() as Dyn<fn(&'static ())>` and `() as Dyn<for<'a> fn(&'a ())>` can have
+// different entries and, because in the former the entry for `foo` is vacant, this test will
+// segfault at runtime.
+
+trait Dyn<U> {
```

Comment:
That's a confusing trait name.^^ Even `Trait` would be better...

Reply:
renamed to `Trait`